### PR TITLE
Add automation for tMFA

### PR DESCRIPTION
**Pull request by moisesolmedo-okta**
### Steps to reproduce:

Actual result:

Expected result:

### CODE REVIEW

File: Sources/DeviceAuthenticator/Transactions/PushChallenge.swift [hunk 1]
```diff
@@ -53,7 +53,7 @@ class PushChallenge: PushChallengeProtocol, CIBAChallengeProtocol {
 
     /// Required if transactionType = CIBA
     lazy var cibaBindingMessage: String? = {
-        return challengeContext["bindingMessage"] as? String
+        return sanitizedBindingMessage
```
Comment (IldarAbdullin-okta) [inline Sources/DeviceAuthenticator/Transactions/PushChallenge.swift:56]: Is it some bug fix?

File: Tests/DeviceAuthenticatorFunctionalTests/VerifyFlowTests.swift [hunk 2]
```diff
@@ -525,6 +525,60 @@ class VerifyFlowTests: XCTestCase {
         }
     }
 
+    func testResolveRetrievePushChallenge_CIBA_Success() {
+
+        let retrivePushChallengeResolveSuccessExpectation = expectation(description: "Retrieve push challenge should resolve")
```
Comment (IldarAbdullin-okta) [inline Tests/DeviceAuthenticatorFunctionalTests/VerifyFlowTests.swift:530]:
```suggestion
        let retrievePushChallengeResolveSuccessExpectation = expectation(description: "Retrieve push challenge should resolve")
```

File: Tests/DeviceAuthenticatorFunctionalTests/VerifyFlowTests.swift [hunk 3]
```diff
@@ -525,6 +525,60 @@ class VerifyFlowTests: XCTestCase {
...
+        // retrive push challenge
+        let retriveJWT = FakePushChallenge.mockIDXJWT(enrollmentId: "aen1jisLwwTG7qRrH0g4", transactionType: "CIBA", bindingMessage: "Did you approve this transaction?")
```
Comment (IldarAbdullin-okta) [inline Tests/DeviceAuthenticatorFunctionalTests/VerifyFlowTests.swift:532]:
```suggestion
        let retrieveJWT = FakePushChallenge.mockIDXJWT(enrollmentId: "aen1jisLwwTG7qRrH0g4", transactionType: "CIBA", bindingMessage: "Did you approve this transaction?")
```

File: Tests/DeviceAuthenticatorFunctionalTests/VerifyFlowTests.swift [hunk 4]
```diff
@@ -525,6 +525,60 @@ class VerifyFlowTests: XCTestCase {
...
+        let retriveChallengeInfo = [["payloadVersion": "IDXv1", "challenge": retriveJWT]]
```
Comment (IldarAbdullin-okta) [inline Tests/DeviceAuthenticatorFunctionalTests/VerifyFlowTests.swift:533]:
```suggestion
        let retrieveChallengeInfo = [["payloadVersion": "IDXv1", "challenge": retriveJWT]]
```

**Review by IldarAbdullin-okta — CHANGES_REQUESTED** [review]
Overall looks good!

Please prepare automation with MyAccount API as well

File: Sources/DeviceAuthenticator/Transactions/PushChallenge.swift (hunk 1 above)
Comment (moisesolmedo-okta) [reply Sources/DeviceAuthenticator/Transactions/PushChallenge.swift:56]: it's this ticket:
https://oktainc.atlassian.net/browse/OKTA-545093
backend is not doing any sanitization on their end. One of the test cases was to use special characters in the binding message. @FeiChen-okta

Comment (IldarAbdullin-okta) [reply Sources/DeviceAuthenticator/Transactions/PushChallenge.swift:56]: Got it. Recommendation for future is to create separate PR for each jira

**Comment by moisesolmedo-okta** [comment]
@IldarAbdullin-okta Should I do myAccount automation on this PR?

**Comment by IldarAbdullin-okta** [comment]
> @IldarAbdullin-okta Should I do myAccount automation on this PR?

We can add in another PR. But our main API is MyAccount API so we should prioritize it first for future automations

**Comment by moisesolmedo-okta** [comment]
@IldarAbdullin-okta Yeah I can see all of our automation uses Legacy API. Will start with the changes on a different PR.

**Review by IldarAbdullin-okta — APPROVED** [review]
LGTM!

**Pull request merged**